lib/cylc/config.py

```diff
@@ -1590,9 +1590,7 @@ def generate_edges(self, lexpression, left_nodes, right, seq,
                     # (TODO - but not explicit null offsets like [-P0D]!)
                     if suicide:
                         # Self-suicide may be OK.
-                        print >> sys.stderr, (
-                            'WARNING: self-suicide is not recommended: '
-                            '%s => !%s.' % (left, right))
+                        continue
```

Comment:
This be `pass`, not `continue`.  As is, it changes the logic so that self-suicide edges are no longer added to the graph structure (only affects visualization, I think).   I guess we need another test then, to compare `cylc graph` output on a suite with a self-suicide in it.